### PR TITLE
Proper handling of epoll_ctl errors

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

When using epoll_ctl we should respect the return value and do the right thing depending on it.

Modifications:

Adjust java and native code to respect epoll_ctl return values.

Result:

Correct and cleaner code.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@trustin please check

File: transport-native-epoll/src/main/c/io_netty_channel_epoll_Native.h [hunk 1]
```diff
@@ -38,9 +38,9 @@ void Java_io_netty_channel_epoll_Native_eventFdWrite(JNIEnv* env, jclass clazz,
 void Java_io_netty_channel_epoll_Native_eventFdRead(JNIEnv* env, jclass clazz, jint fd);
 jint Java_io_netty_channel_epoll_Native_epollCreate(JNIEnv* env, jclass clazz);
 jint Java_io_netty_channel_epoll_Native_epollWait0(JNIEnv* env, jclass clazz, jint efd, jlong address, jint length, jint timeout);
-void Java_io_netty_channel_epoll_Native_epollCtlAdd(JNIEnv* env, jclass clazz, jint efd, jint fd, jint flags);
-void Java_io_netty_channel_epoll_Native_epollCtlMod(JNIEnv* env, jclass clazz, jint efd, jint fd, jint flags);
-void  Java_io_netty_channel_epoll_Native_epollCtlDel(JNIEnv* env, jclass clazz, jint efd, jint fd);
+jint Java_io_netty_channel_epoll_Native_epollCtlAdd0(JNIEnv* env, jclass clazz, jint efd, jint fd, jint flags);
+jint Java_io_netty_channel_epoll_Native_epollCtlMod0(JNIEnv* env, jclass clazz, jint efd, jint fd, jint flags);
+jint  Java_io_netty_channel_epoll_Native_epollCtlDel0(JNIEnv* env, jclass clazz, jint efd, jint fd);
```
Comment (trustin) [inline transport-native-epoll/src/main/c/io_netty_channel_epoll_Native.h]: double space -> single space

**Comment by trustin** [comment]
LGTM. Please cherry-pick after addressing the comment.

**Comment by normanmaurer** [comment]
Addressed comment and cherry-picked into 4.0 (588f5c2476c08455ff3ecaa562ef0d5015b84664) , 4.1 (5f5cdd10895f6ee4800104d1b95e2960e2b2325e) and master (b3abd58b05a4d73b4ba50f662fd45f9a73ff93ba)

**Pull request closed without merging**